src/librustc_interface/interface.rs

```diff
@@ -143,17 +144,19 @@ pub struct Config {
         Option<fn(&Session, &mut ty::query::Providers<'_>, &mut ty::query::Providers<'_>)>,
 }
 
-pub fn run_compiler_in_existing_thread_pool<F, R>(config: Config, f: F) -> R
-where
-    F: FnOnce(&Compiler) -> R,
-{
+pub fn run_compiler_in_existing_thread_pool<R>(
+    config: Config,
+    registry: Registry,
```

Comment:
Could this be part of `Config`? It feels a bit odd to have it separate given that it's both immutable and essentially configuration...

Reply:
Sure; moved over -- this caused some slight duplication but I think it makes sense still.